pb/lmsysmd/dataset/v1/dataset.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package lmsysmd.dataset.v1;
+
+import "buf/validate/validate.proto";
+import "google/protobuf/timestamp.proto";
+
+service DatasetService {
+  rpc CreateDataset(CreateDatasetRequest) returns (CreateDatasetResponse) {
+    option idempotency_level = IDEMPOTENT;
```

Comment:
this is not idempotent as it causes an autoincrement to the database id column upon insertion

Reply:
Ok I think I don't have a full understanding of idempotency in this context - my thought process here is that if you try to create the same dataset it shouldn't do anything or else you would have duplicates. It shouldn't even autoincrement it no?

Reply:
sounds good, would you please add a comment here describing this guarantee?